test/test_capture_prefetch.cpp

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ * Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "gtest/gtest.h"
+
+#include "logging/tf_graph_writer.h"
+#include "ngraph_bridge/ngraph_api.h"
+#include "ngraph_bridge/ngraph_capture_variables.h"
+#include "ngraph_bridge/ngraph_utils.h"
+#include "ngraph_bridge/version.h"
+#include "test/test_utilities.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+namespace testing {
+
+TEST(CapturePrefetchTest, SmallGraph1) {
+  list<string> env_vars{"NGRAPH_TF_USE_PREFETCH"};
+  const unordered_map<string, string>& env_map = StoreEnv(env_vars);
+  SetEnvVariable("NGRAPH_TF_USE_PREFETCH", "1");
+  GraphConstructorOptions opts;
+  opts.allow_internal_ops = true;
+  Graph input_graph(OpRegistry::Global());
+
+  // Now read the graph
+  // test_capture_prefetch.pbtxt was created by running the following
+  // command:
+  // NGRAPH_TF_DUMP_GRAPHS=1 python examples/axpy_pipelined.py
+  // and using the precapture_0000.pbtxt
+  ASSERT_OK(LoadGraphFromPbTxt("test_capture_prefetch.pbtxt", &input_graph));
+  std::set<string> skip_these_nodes = {};
+  ASSERT_OK(CaptureVariables(&input_graph, skip_these_nodes));
+  int count_ng_prefetch = 0;
+  int count_tf_prefetch = 0;
+  for (auto node : input_graph.op_nodes()) {
+    if (node->type_string() == "NGraphPrefetchDataset") {
+      count_ng_prefetch = count_ng_prefetch + 1;
```

Comment:
does this op have more than 1 outputs? confused by the assert in the other test and not here

Reply:
In the other test since I was looking at the output of the node to make sure it is MakeIterator, I added that extra ASSERT for the num outputs. It can be added here as well.